### ISSUE TITLE
S3353: Error message is unprecise

### ISSUE THREAD

**Issue by martin-strecker-sonarsource** [SonarSource/sonar-dotnet#5868]
### Description

The error message for S3353 is insufficient.

### Repro steps

In a project with C# 9 or below, add this code
```CS
public class Test {
    public void M()
    {
        var message = $"This is a {nameof(Test)}";
    }
}
```
The message shown is `Add the 'const' modifier to 'message'.`

Two actions are missing, which the user has to take besides adding `const`:

### var  must be replaced

`var` must be replaced with the concrete type. Otherwise `CS0822: Implicitly-typed variables cannot be constant` is raised.

```CS
public class Test {
    public void M()
    {
        const string message = $"This is a {nameof(Test)}";
    }
}
```
A better error message would be:

`Add the 'const' modifier to 'message' and replace 'var' with 'string'.`

### Interpolated const strings are only supported in C# 10 and above

C# 10 allows [interpolated strings in conjunction with `const`](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/proposals/csharp-10.0/constant_interpolated_strings). As a workaround for lower versions, the code needs to be rewritten:

```CS
public class Test {
    public void M()
    {
        const string message = "This is a " + nameof(Test);
    }
}
```
A better error message would be:
`Add the 'const' modifier to 'message', replace 'var' with 'string', and use string concatenation instead of an interpolated string.`

The rspec should be updated with more non-compliant and compliant samples to illustrate the work needed.
